mount/mount_linux.go

```diff
@@ -289,55 +288,53 @@ func (mounter *SafeFormatAndMount) formatAndMount(source string, target string,
 		}
 	}
 
-	// Try to mount the disk
-	klog.V(4).Infof("Attempting to mount disk: %s %s %s", fstype, source, target)
-	mountErr := mounter.Interface.Mount(source, target, fstype, options)
-	if mountErr != nil {
-		// Mount failed. This indicates either that the disk is unformatted or
-		// it contains an unexpected filesystem.
-		existingFormat, err := mounter.GetDiskFormat(source)
-		if err != nil {
-			return err
+	// Check if the disk is already formatted
+	existingFormat, err := mounter.GetDiskFormat(source)
+	if err != nil {
+		return err
+	}
+
+	// Use 'ext4' as the default
+	if len(fstype) == 0 {
+		fstype = "ext4"
```

Comment:
here we are saying if user does not specify a fstype, by default we set it "ext4". But why not we can set it as the current existing file system?

Reply:
fstype is the default value for all. means that if existing format is ext3, and fstype is empty, the device will be mounted as ext4, not ext3.